TypeCobol/Compiler/Types/BuiltinTypes.cs

```diff
@@ -0,0 +1,168 @@
+        public readonly static Type UsagePackedDecimalType = new Type(Tags.Usage, UsageFormat.PackedDecimal);
+        public readonly static Type UsageProcedurePointerType = new Type(Tags.Usage, UsageFormat.ProcedurePointer);
+        public readonly static Type UsageComp3Type = new Type(Tags.Usage, UsageFormat.Comp3);
+        #endregion
+
+
+        /// <summary>
+        /// Static constructor
+        /// </summary>
+        static BuiltinTypes()
+        {
+            Comp1Type = new Type(Type.Tags.Usage, Type.UsageFormat.Comp1);
+            FloatType = Comp1Type;
+            FloatType.SetFlag(Symbol.Flags.BuiltinType, true);
+            Comp2Type = new Type(Type.Tags.Usage, Type.UsageFormat.Comp2);
+            DoubleType = Comp2Type;
+            DoubleType.SetFlag(Symbol.Flags.BuiltinType, true);
+            PointerType = new Type(Type.Tags.Usage, Type.UsageFormat.Pointer);
+            PointerType.SetFlag(Symbol.Flags.BuiltinType, true);
+            FunctionPointerType = new Type(Type.Tags.Usage, Type.UsageFormat.FunctionPointer);
+            FunctionPointerType.SetFlag(Symbol.Flags.BuiltinType, true);
+            ProcedurePointerType = new Type(Type.Tags.Usage, Type.UsageFormat.ProcedurePointer);
+            ProcedurePointerType.SetFlag(Symbol.Flags.BuiltinType, true);
+            IndexType = new Type(Type.Tags.Usage, Type.UsageFormat.Index);
+            IndexType.SetFlag(Symbol.Flags.BuiltinType, true);
+            ObjectReferenceType = new Type(Type.Tags.Usage, Type.UsageFormat.ObjectReference);
+            ObjectReferenceType.SetFlag(Symbol.Flags.BuiltinType, true);
+
+            OmittedType = new Type(Type.Tags.Usage, Type.UsageFormat.Omitted);
+            OmittedType.SetFlag(Symbol.Flags.BuiltinType, true);
+            AlphabeticType = new Type(Type.Tags.Usage, Type.UsageFormat.Alphabetic);
+            AlphabeticType.SetFlag(Symbol.Flags.BuiltinType, true);
+            NumericType = new Type(Type.Tags.Usage, Type.UsageFormat.Numeric);
+            NumericType.SetFlag(Symbol.Flags.BuiltinType, true);
+            NumericEditedType = new Type(Type.Tags.Usage, Type.UsageFormat.NumericEdited);
+            NumericEditedType.SetFlag(Symbol.Flags.BuiltinType, true);
+            AlphanumericType = new Type(Type.Tags.Usage, Type.UsageFormat.Alphanumeric);
+            AlphanumericType.SetFlag(Symbol.Flags.BuiltinType, true);
+            AlphanumericEditedType = new Type(Type.Tags.Usage, Type.UsageFormat.AlphanumericEdited);
+            AlphanumericEditedType.SetFlag(Symbol.Flags.BuiltinType, true);
+            DBCSType = new Type(Type.Tags.Usage, Type.UsageFormat.DBCS);
+            DBCSType.SetFlag(Symbol.Flags.BuiltinType, true);
+            FloatingPointType = new Type(Type.Tags.Usage, Type.UsageFormat.FloatingPoint);
+            FloatingPointType.SetFlag(Symbol.Flags.BuiltinType, true);
+            OccursType = new Type(Type.Tags.Usage, Type.UsageFormat.Occurs);
```

Comment:
Why Type.Tags.Usage here and not Type.Tags.Array ?

Reply:
Ok this is my translation of the TypeCobol Builtin Type Occurs in the file DataType.cs
public static readonly DataType Occurs             = new DataType("Array");